### ISSUE TITLE
Recurrence count in AgentAppointmentEdit

### ISSUE THREAD

**Issue by bschmalhofer** [RotherOSS/otobo#2035]
When adding a recurring appointment one can set the number of occurrences. It is obvious that this should be a positive integer. But the GUI allows to enter garbage. The effect is that the appointment is not created.

TODO:

- [ ] add a JavaScript Check of the input value

**Comment by stefanhaerter** [comment]
Would it also be meaningful to check the value server-side?

**Comment by bschmalhofer** [comment]
Yes, likely communicating the failed AppointmentCreate() or AppointmentUpdate() to the frontend module.

**Comment by stefanhaerter** [comment]
I just recognized that the RecurrenceCount has a maxlength of 2. I would assume that this is intended.

**Comment by bschmalhofer** [comment]
According to https://github.com/OTRS/otrs/commit/4784db74b6da018d7eeaccae25a841d9d4ac67f4 the maximal length of 2 already was in the package  OTRSAppointmentCalendar. But in the database the field calendar_appointment.recur_count is an integer. I suppose the limit 2 is mostly accidental.

**Comment by stefanhaerter** [comment]
On the HTML side, we could use:

```html
<input type="number" min="0">
```

**Comment by stefanhaerter** [comment]
Putting this in the OTOBO Issues project for planning.

**Comment by stefanhaerter** [comment]
PR merged into 11.1. Closing issue and updating milestone.

**Comment by MichaelThumes** [comment]
pls see notes in linked bug ticket for reasons for re-opening
